Mathlib/Topology/Order/Category/FrameAdjunction.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2023 Anne Baanen, Sam v. Gool, Leo Mayer, Brendan S. Murphy. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Sam v. Gool, Leo Mayer, Brendan S. Murphy
+-/
+import Mathlib.Topology.Category.Locale
+
+/-!
+# Adjunction between Locales and Topological Spaces
+
+This file defines functors between the categories of Locales and Topological Spaces
+and proves that they form an adjunction.
+
+## Main declarations
+
+* `pt`: the *points* functor from the category of locales to the category of topological spaces.
+* `topToLocale`: the forgetful functor from the category of topological spaces to the category of
+  locales.
+
+- `locale_top_adjunction`: the theorem that topToLocale is left adjoint to pt.
```

Comment:
As well as fully qualifying the names for docs, I suggest not to use the word "theorem" since there's data here (indeed it's a `def`):
```suggestion
- `Locale.adjunctionTopToLocalePT`: the left adjunction between `topToLocale` and `Locale.pt`.
```

Reply:
OK. I rephrased your suggestion slightly ("left adjunction" sounds odd to me).